front-end/src/components/analysis-page/IntentDiagram.jsx

```diff
@@ -1,17 +1,52 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import store from '../../store.js';
-import { toggleFlag, toggleStar } from '../../store/analyzeTranscriptSlice.js';
+import { goToIntentByQuestion, toggleFlag, toggleStar } from '../../store/analyzeTranscriptSlice.js';
 import StarAPI from "../../services/StarAPI.js";
 import FlagAPI from "../../services/FlagAPI.js";
 
 class IntentDiagram extends Component {
+
+    generateIntentBoxStyling = (flash) => {
+        // Base styling
+        let styling = "rounded-lg p-5 mt-9 shadow-lg shadow-blue/10 break-words text-center text-lg 2xl:text-2xl w-52 sm:min-w-[30rem] sm:max-w-[35rem] 2xl:max-h-[14rem] ";
+        // Add color based on whether the box is flashing
+        styling += flash ? "bg-green-100" : "bg-off-white";
+        // Add any previous intents merged into this intent, if any
+        if (this.props.previousIntents) {
+
+        }
+        return styling
+    }
+
     listLeaves = () => {
         return Object.keys(this.props.children).map((key) => {
+            const isEndOfConversation = (key === "END OF CONVERSATION");
```

Comment:
Perhaps this needs to change to
```suggestion
            const isEndOfConversation = (key === "[END OF CONVERSATION]");
```